pkg/storage/bench_test.go

```diff
@@ -1526,31 +1539,41 @@ func runMVCCGarbageCollect(
 	ts := hlc.Timestamp{}.Add(time.Date(2000, 1, 1, 0, 0, 0, 0, time.UTC).UnixNano(), 0)
 	val := roachpb.MakeValueFromBytes(randutil.RandBytes(rng, opts.valueBytes))
 
-	// We write values at ts+(0,i), set now=ts+(1,0) so that we're ahead of all
+	// We write values at ts+(1,i), set now=ts+(2,0) so that we're ahead of all
 	// the writes. This value doesn't matter in practice, as it's used only for
 	// stats updates.
-	now := ts.Add(1, 0)
+	now := ts.Add(2, 0)
 
 	// Write 'numKeys' of the given 'keySize' and 'valSize' to the given engine.
 	// For each key, write 'numVersions' versions, and add a GCRequest_GCKey to
 	// the returned slice that affects the oldest 'deleteVersions' versions. The
-	// first write for each key will be at `ts`, the second one at `ts+(0,1)`,
-	// etc.
+	// first write for each key will be at `ts+(1,0)`, the second one
+	// at `ts+(1,1)`, etc.
+	// If numRangeKeys is set to 1 then range tombstone will be written at ts.
 	//
 	// NB: a real invocation of MVCCGarbageCollect typically has most of the keys
 	// in sorted order. Here they will be ordered randomly.
 	setup := func() (gcKeys []roachpb.GCRequest_GCKey) {
 		batch := eng.NewBatch()
+		if opts.numRangeKeys > 1 {
+			b.Fatal("Invalid bench data config. Number of range keys can be 0 or 1")
+		}
+		if opts.numRangeKeys == 1 {
+			if err := MVCCDeleteRangeUsingTombstone(ctx, batch, nil, keys.MinKey.Next(), keys.MaxKey,
```

Comment:
We don't really allow MVCC range tombstones across local keyspace, but the check is currently a bit loose. Added #87403 to tighten it up. This call should use `Keys.LocalMax` as the start key.